### PR TITLE
Consolidated project structure

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

EFAX-11,22,24,25

This large PR is a consolidation effort to merge all project modules into the main project - it is purely a refactoring of code.  

As the project grew, it became evident that having separate java projects for each module (mailservice, mail-listener, webservice, etc) exponentially complicated the project with inter-module communication and duplicate code.   There is now only one project, jag-efax, which significantly simplifies docker-compose and project infrastructure. Each of the modules have their own logical domain separation in the main project code base.

I tried to keep code changes to a minimum so most of these files changes are simply from moving a file from one folder to another - java import statement changes.

`docker-compose up -d` will bring up 3 containers, webservice, redis, and redis-commander.
- webservice - this SOAP web application, available at `http://localhost:8080/ws`
- redis - a key-value persistent store
- redis-commander - a UI for redis

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

`mvn verify`

`docker-compose up -d`
`http://localhost:8080/actuator/health` returns status:UP

## Does the change impact or break the Docker build?

- [x] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [x] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java [hunk 1]
```diff
@@ -55,29 +55,30 @@
...
+import ca.bc.gov.ag.efax.mail.repository.SentMessageRepository;
+import ca.bc.gov.ag.efax.mail.util.StringUtils;
+import ca.bc.gov.ag.efax.pdf.service.PdfService;
+import ca.bc.gov.ag.efax.ws.exception.FAXSendFault;
 import ca.bc.gov.jag.ews.proxy.ExchangeWebServiceClient;
 
 @Service
-public class MailService {
-
+// FIXME: techdept - codeclimate reports this class is too large.
```
Comment (BrendanBeachBC) [inline src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java:68]: Small typo in comment
```suggestion
// FIXME: techdebt - codeclimate reports this class is too large.
```

File: src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java [hunk 2]
```diff
@@ -87,20 +88,21 @@ public void sendMessage(final MailMessage mailMessage) throws MailException {
...
-            throw new MailException("Unknown Exception in class sendMessage", e);
+            throw new FAXSendFault("Unknown Exception in class sendMessage", e);
         }
     }
 
 	// Post processing cleanup
 	// Delete temp attachment(s) file(s)
 	// Remove uuid-message key-value pair from messageMap
 	private void cleanupMessage(MailMessage msg) throws Exception {
+	    // FIXME: techdept - codeclimate reports this class is too complex.
```
Comment (BrendanBeachBC) [inline src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java:99]:
```suggestion
	    // FIXME: techdebt - codeclimate reports this class is too complex.
```

File: src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java [hunk 3]
```diff
@@ -139,32 +141,34 @@ private File readFileFromURL(final String url, final String outFilename) throws
...
+			throw new Exception("Unknown Exception in class readFileFromURL", e);
 		} finally {
 			// NOP
 		}
 	}
 
-	private boolean processMessage(MailMessage m) throws MailException {
+	private boolean processMessage(MailMessage m) throws Exception {
+        // FIXME: techdept - codeclimate reports this class is too complex.
+        // FIXME: techdept - codeclimate reports this class is too large
```
Comment (BrendanBeachBC) [inline src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java:158]:
```suggestion
        // FIXME: techdebt - codeclimate reports this class is too complex.
        // FIXME: techdebt - codeclimate reports this class is too large
```

**Review by karthikmsqa — APPROVED** [review]
Excellent @KDNeufeld!

**Review by BrendanBeachBC — COMMENTED** [review]
Looks good to me too, just wanted to mention a couple small comment typos that caught my eye. Nothing major

File: src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java (hunk 1 above)
Comment (KDNeufeld) [reply src/main/java/ca/bc/gov/ag/efax/mail/service/MailService.java:68]: :) Fantastic.  I'll make these changes to my next PR.
Thanks!

**Pull request merged**